### PR TITLE
Feature/summary view

### PR DESCRIPTION
**Pull request by flfue**
Adding summary table to provide a compact overview
- grouping by effect and category
- links to jump directly to the particular fishbone view
- filter for each element 

image

### CODE REVIEW

**Comment by mbehr1** [comment]
nice feature! really useful!

File: src/webview/src/App.js [hunk 1]
```diff
@@ -222,6 +223,9 @@ export default class App extends Component {
       this.state.fbPath = [{ title: this.state.title, effectIndex: 0 }]
     }
 
+    // Ensure summary dialog is initially disabled
+    this.setState({ showSummaryDialog: false })
```
Comment (mbehr1) [inline src/webview/src/App.js:227]: I think that in constructor (and only there) you should modify this.state directly. e.g. 
```
this.state['showSummaryDialog']=false;
```

File: src/webview/src/components/summary/dataProvider.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            disableGroupBy: true,
+          },
+        ],
+      },
+    ],
+    []
+  )
+}
+
+var FbPathChangeHook = undefined;
```
Comment (mbehr1) [inline src/webview/src/components/summary/dataProvider.js:182]: could be inside the function "SummaryDataProvider"?

File: src/webview/src/components/summary/table.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            return (
+              <React.Fragment>
+                <TableRow hover={true} {...row.getRowProps()}>
+                  {row.cells.map(cell => {
+                    return (
+                      <TableCell className={classes.tableCell} style={{ whiteSpace: 'pre-line' }} {...cell.getCellProps()}>
+                        {cell.isGrouped ? (
+                          <div>
+                            <span {...row.getToggleRowExpandedProps()}>
+                              {row.isExpanded ? '➖' : '➕'}
```
Comment (mbehr1) [inline src/webview/src/components/summary/table.js:203]: the '-' and '+' are some special UTF-8 chars? (not the usual -/+ or?)

File: src/webview/src/components/summaryDialog.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import Dialog from '@material-ui/core/Dialog';
+import { DialogContent } from '@material-ui/core';
+
+import SummaryTable from './summary/table';
+import { SummaryHeaderProvider, SummaryDataProvider } from './summary/dataProvider';
+
+/**
+ * Planned features:
+ * - Consider badges with filters and results
+ * - Export as PDF
```
Comment (mbehr1) [inline src/webview/src/components/summaryDialog.js:12]: +1 :-)

**Review by mbehr1 — COMMENTED** [review]
cool, looks good! will test on target, code looks good (just minor comments/questions).

File: src/webview/src/components/summary/table.js (hunk 3 above)
Comment (flfue) [reply src/webview/src/components/summary/table.js:203]: Yes it's [U+2795](https://unicode.org/emoji/charts/full-emoji-list.html#2795) and [U+2796](https://unicode.org/emoji/charts/full-emoji-list.html#2796) which is from the Emoji List and used as very simple buttons on the table.

File: src/webview/src/components/summary/dataProvider.js (hunk 2 above)
Comment (flfue) [reply src/webview/src/components/summary/dataProvider.js:182]: I agree.

File: src/webview/src/App.js (hunk 1 above)
Comment (flfue) [reply src/webview/src/App.js:227]: I agree.

File: src/webview/src/App.js [hunk 5]
```diff
@@ -224,7 +224,7 @@ export default class App extends Component {
     }
 
     // Ensure summary dialog is initially disabled
-    this.setState({ showSummaryDialog: false })
+    this.setState['showSummaryDialog'] = false;
```
Comment (mbehr1) [inline src/webview/src/App.js:227]: this.state... , or?

Comment (mbehr1) [reply src/webview/src/App.js:227]: I'll fix that during merge / update of version...

**Review by mbehr1 — APPROVED** [review]

**Comment by mbehr1** [comment]
pretty cool! thx a lot!

**Pull request merged**